src/fable-library/Array.fs

```diff
@@ -1416,14 +1416,23 @@ let resize
     if newSize < 0 then
         invalidArg "newSize" "The input must be non-negative."
 
-    let len = xs.Length
+    let len =
+        if isNull xs then
+            if newSize = 0 then
+                xs <- allocateArrayFromCons cons 0
+
+            0
+        else
+            xs.Length
 
```

Comment:
@chkn I know it already works, but perhaps simplify the logic a bit like this (untested):

instead of:
 ```fs
    let len =
        if isNull xs then
            if newSize = 0 then
                xs <- allocateArrayFromCons cons 0
            0
        else
            xs.Length
    ....
```

perhaps this:
```fs
if isNull xs then
    xs <- allocateArrayFromCons cons newSize
else
    let len = xs.Length
....
```